### PR TITLE
Managementgroups/20191101

### PR DESCRIPTION
**Pull request by ramoka178**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 1]
```diff
@@ -0,0 +1,1684 @@
...
+    }
+  },
+  "tags": [
+    {
+      "name": "ManagementGroups",
+      "description": "A Management Group is a customer defined scope (grouping mechanism) that \nprovides access control (authorization), policy management and reporting. \nManagement Groups are organized in a strictly tree-based hierarchy.\n"
+    },
+    {
+      "name": "Operations",
+      "description": "Management operations supported by the Microsoft.Management resource provider.\n"
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: Why do all of the descriptions need a newline at the end?

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 2]
```diff
@@ -0,0 +1,1684 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "GetManagementGroup": {
+            "$ref": "./examples/GetManagementGroup.json"
+          },
+          "GetManagementGroupWithExpand": {
+            "$ref": "./examples/GetManagementGroupWithExpand.json"
+          },
+          "GetManagementGroupsWithExpandAndResurse": {
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: Recurse

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 3]
```diff
@@ -0,0 +1,1684 @@
...
+        },
+        "x-ms-examples": {
+          "DeleteManagementGroup": {
+            "$ref": "./examples/DeleteManagementGroup.json"
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{groupId}/descendants" : {
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: extra space before the ':'

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 4]
```diff
@@ -0,0 +1,1684 @@
...
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "GetEntities": {
+            "$ref": "./examples/GetEntities.json"
+          }
+        },
+        "x-ms-long-running-operation": true
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: If this is a long running operation it should have a 202 status code option

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 5]
```diff
@@ -0,0 +1,1684 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "StartTenantBackfill": {
+            "$ref": "./examples/StartTenantBackfillRequest.json"
+          }
+        },
+        "x-ms-long-running-operation": true
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: Also needs a 202 response if this is async

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 6]
```diff
@@ -0,0 +1,1684 @@
...
+        "children": {
+          "description": "The list of children.",
+          "x-nullable": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagementGroupChildInfo"
+          }
+        },
+        "path": {
+          "description": "The path from the root to the current group.",
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: What order is this in?  Top to bottom or bottom to top?  Description should be explicit

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 7]
```diff
@@ -0,0 +1,1684 @@
...
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "the name to check for availability"
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "/providers/Microsoft.Management/managementGroup"
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: "managementGroups", here and 1495

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 8]
```diff
@@ -0,0 +1,1684 @@
...
+    },
+    "CheckNameAvailabilityParameter": {
+      "name": "checkNameAvailabilityRequest",
+      "in": "body",
+      "required": true,
+      "description": "Management group name availability check parameters.",
+      "schema": {
+        "$ref": "#/definitions/CheckNameAvailabilityRequest"
+      },
+      "x-ms-parameter-location": "client"
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: why were these moved off of method to client?

File: specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON [hunk 9]
```diff
@@ -0,0 +1,1684 @@
...
+          "readOnly": true
+        },
+        "status": {
+          "type": "string",
+          "description": "The current status of the operation",
+          "readOnly": true
+        }
+      }
+    },
+    "AzureAsyncOperationResults" : {
```
Comment (pilor) [inline specification/managementgroups/resource-manager/Microsoft.Management/stable/2019-11-01/management.JSON]: where is this used?

**Review by pilor — CHANGES_REQUESTED** [review]

**Comment by ramoka178** [comment]
@sergey-shandar : Could you provide any inputs as why the checks failed even when loading the swagger json file ?

**Comment by pilor** [comment]
Signing off from ARM side.  No real changes from the existing preview version.

**Review by pilor — APPROVED** [review]

**Review by sergey-shandar — APPROVED** [review]
